src/z3c/pt/tests/test_templates.py

```diff
@@ -118,19 +124,107 @@ def render(self):
                 implementedBy(View)
-                ),
+            ),
             IContentProvider,
             name="content"
-            )
+        )
 
         context = Context()
         request = Request()
 
         result = view(context=context, request=request)
-        self.assertTrue(repr(data) in result)
-        self.assertTrue(repr({'context': context}) in result)
+        self.assertIn(repr(data), result)
+        self.assertIn(repr({'context': context}), result)
+
+class TestOpaqueDict(unittest.TestCase):
+
+    def test_getitem(self):
+        import operator
+        d = {}
+        od = pagetemplate.OpaqueDict(d)
+        with self.assertRaises(KeyError):
+            operator.itemgetter('key')(od)
+
+        d['key'] = 42
+        self.assertEqual(od['key'], 42)
+
+    def test_len(self):
+        d = {}
+        od = pagetemplate.OpaqueDict(d)
+        self.assertEqual(0, len(od))
+
+        d['key'] = 42
+        self.assertEqual(1, len(od))
+
+    def test_repr(self):
+        d = {}
+        od = pagetemplate.OpaqueDict(d)
+        self.assertEqual('{...} (0 entries)', repr(od))
+
+        d['key'] = 42
+        self.assertEqual('{...} (1 entries)', repr(od))
+
+
+class TestBaseTemplate(unittest.TestCase):
+
+    def test_negotiate_fails(self):
+        class I18N(object):
+            request = None
+            def negotiate(self, request):
+                self.request = request
+                raise Exception("This is caught")
+
+        i18n = I18N()
+        orig_i18n = pagetemplate.i18n
+        pagetemplate.i18n = i18n
+        try:
+            template = pagetemplate.BaseTemplate('<html />')
+            request = 'strings are allowed'
+            template.render(request=request)
+            self.assertIs(i18n.request, request)
+        finally:
+            pagetemplate.i18n = orig_i18n
+
+    def test_translate_mv(self):
+        template = pagetemplate.BaseTemplate("""
+        <html>
+          <body metal:use-macro="m" />
+        </html>
+        """)
+
+        class Macro(object):
+            translate = None
+            def include(self, stream, econtext, *args, **kwargs):
+                self.translate = econtext['translate']
+        macro = Macro()
+        template.render(m=macro)
+
+        self.assertIsNone(macro.translate(pagetemplate.MV))
+
+class TestBaseTemplateFile(unittest.TestCase):
+
+    def test_init_with_path(self):
+
+        here = os.path.abspath(os.path.dirname(__file__))
+
+        template = pagetemplate.BaseTemplateFile('view.pt', path=here)
+
+        self.assertEqual(template.filename,
+                         os.path.join(here, 'view.pt'))
+
+class TestBoundPageTemplate(unittest.TestCase):
+
+    def test_setattr(self):
+        bound = pagetemplate.BoundPageTemplate(None, None)
+        with self.assertRaisesRegexp(AttributeError,
+                                     "Can't set attribute"):
+            setattr(bound, 'im_self', 42)
```

Comment:
Again, `bound.im_self = 42` would be more straightforward.

Reply:
To my eyes, the function call of `setattr` emphasises what's going on. It looks "stronger" than simply assigning an attribute.

Reply:
This is not worth bickering about, even if you're obviously wrong here ;-)